app/grandchallenge/evaluation/views/__init__.py

```diff
@@ -1252,3 +1255,94 @@ def get_object(self, queryset=None):
             challenge=self.request.challenge,
             slug=self.kwargs["slug"],
         )
+
+
+class AlgorithmInterfaceForPhasePermissionMixin(AccessMixin):
+    @property
+    def phase(self):
+        return get_object_or_404(
+            Phase,
+            challenge=self.request.challenge,
+            challenge__phase__submission_kind=SubmissionKindChoices.ALGORITHM,
+            challenge__phase__external_evaluation=False,
+            slug=self.kwargs["slug"],
+        )
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_staff:
```

Comment:
I think we want a finer grained permission for this as probably not all staff users should be able to do it.

Reply:
Ok, fair point. We could use the existing `configure_algorithm_phase` permission for that. This permission is currently assigned to the RSE group in the Django admin. Or do you think this requires a new permission just for Chris, Miriam, you and me?

Reply:
Using `configure_algorithm_phase` makes sense to me given that this is the permission that is required to change a phase into an algorrithm phase as well. This is a related action I would say.

Reply:
That sounds like the perfect one!